src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/PgScheduledEmailManager.java

```diff
@@ -0,0 +1,67 @@
+package uk.ac.cam.cl.dtg.isaac.api.managers;
+
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.ac.cam.cl.dtg.segue.dao.SegueDatabaseException;
+import uk.ac.cam.cl.dtg.segue.database.PostgresSqlDb;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.ZonedDateTime;
+
+public class PgScheduledEmailManager {
+    private static final Logger log = LoggerFactory.getLogger(PgScheduledEmailManager.class);
+
+    private final PostgresSqlDb database;
+
+    /**
+     * Creates a new user data manager object.
+     *
+     * @param database
+     *            - the database reference used for persistence.
+     */
+    @Inject
+    public PgScheduledEmailManager(final PostgresSqlDb database) {
+        this.database = database;
+    }
+
+    public Boolean scheduledEmailAlreadySent(String emailKey) throws SegueDatabaseException {
+        String query = "SELECT EXISTS(SELECT 1 FROM scheduled_emails WHERE email_id = ?)";
+        try (Connection conn = database.getDatabaseConnection();
+             PreparedStatement pst = conn.prepareStatement(query);
+        ) {
+            pst.setString(1, emailKey);
+            try (ResultSet results = pst.executeQuery()) {
+                if (results.next()) {
+                    return results.getBoolean(1);
+                } else {
+                    return true;
```

Comment:
Is this right? Don't we only follow this branch if there is no result in the ResultSet? (Which I could believe shouldn't ever be possible, but then we should log an error to be clear something is very wrong?).

Reply:
Yes you're right, this is a catchall 'if for some reason it breaks don't try sending an email' - I'll change it to an error